rust/scx_utils/src/topology.rs

```diff
@@ -483,14 +510,21 @@ fn create_insert_cpu(
         None => CoreType::Big { turbo: false },
     };
 
-    let core = llc_mut.cores.entry(core_id).or_insert(Arc::new(Core {
-        id: core_id,
+    let inc_core = if llc_mut.cores.contains_key(core_id) || !flatten_llc {
+        true
+    } else {
+        false
+    };
```

Comment:
Ditto with cores.

So, for LLCs, there are two problems to solve:
1. LLC IDs may not be consecutive.
2. A single LLC ID may straddle across multiple nodes incorrectly in some virt setups.

The second one probably should be solved by reassigning a different kernel_id to the llc. Let's ignore that for a bit. For the first issue, we should be building kernel ID -> consecutive ID mapping. I don't see how keeping using the ssame ID when the same kernel ID was seen would assign the correct consecutive IDs.

For core IDs, the problem is that they may repeat across node boundaries, so this is a bit different from the LLC situation - ie. we should be assigning each node_id:core_kernel_id pair a unique consecutive ID.

Reply:
> I don't see how keeping using the ssame ID when the same kernel ID was seen would assign the correct consecutive IDs.

It only increments when the mapping hasn't been seen before, but I guess on repeats it would be incorrect. I was trying to avoid having to create that extra mapping, but it looks like it's required. I might leave private to the structs though.

Reply:
We don't have to keep it around either and can just use temporary context struct containing necessary mappings and cursors during init.